### ISSUE TITLE
Get final result with new test image

### ISSUE THREAD

**Issue by NguyenHongSon1103** [Runinho/pytorch-cutpaste#15]
Hi,  
I am confusing about how to determine if an image is anomalous or not because model has two output: classification prediction and GDE scores.  
Should I use a threshold for GDE scores to check and ignore model prediction ?
Thank you for your work and your reply :)

**Comment by Runinho** [comment]
> Should I use a threshold for GDE scores to check and ignore model prediction ?

Yes, during inference/validation the paper uses the Density Estimation to detect anomalous data.
The Model is used as a feature extractor on which the density estimation is performed.

During training the whole model, the part that is used as a feature extractor plus a model head, is trained on the cutpast classification proxy task. So the classification output is only used as a proxy task to train the model.

**Comment by NguyenHongSon1103** [comment]
can you please tell me how to calculate that threshold, i don't see it in your code.

**Comment by Runinho** [comment]
> can you please tell me how to calculate that threshold, i don't see it in your code.

I answered a similar question [here](https://github.com/Runinho/pytorch-cutpaste/issues/9#issuecomment-944344173):
The thresholding in the evaluation is done by calculating the roc. This might be an option for your implementation (See ROC on [wikipedia](https://en.wikipedia.org/wiki/Receiver_operating_characteristic)).

Another Paper by Ripple et al. which Li et. al reference use a probabilistic perspective of calculating thresholds: https://github.com/ORippler/gaussian-ad-mvtec/blob/a2e15800f0087fc7b965e491c53a73f8c7480d1e/src/gaussian/model.py#L152
